contracts/Holdable.sol

```diff
@@ -0,0 +1,225 @@
+pragma solidity ^0.5.0;
+
+import "openzeppelin-solidity/contracts/token/ERC20/ERC20.sol";
+import "./IHoldable.sol";
+import "./libraries/StringUtil.sol";
+
+
+contract Holdable is IHoldable, ERC20 {
+
+    using SafeMath for uint256;
+    using StringUtil for string;
+
+
+    struct Hold {
+        address issuer;
+        address origin;
+        address target;
+        address notary;
+        uint256 expiration;
+        uint256 amount;
+        HoldStatusCode status;
+    }
+
+    mapping(string => Hold) private holds;
+    mapping(address => uint256) private heldBalance;
+    mapping(address => mapping(address => bool)) private operators;
+
+    uint256 private _totalHeldBalance;
+
+    function hold(string calldata operationId, address to, address notary, uint256 value, uint256 timeToExpiration) external returns (bool){
```

Comment:
`hold` and `holdFrom` are almost the same, except for the check of the operator and the value of the issuer.
It's better to move the code in a private function `_hold(string calldata operationId, address issuer, address to, address notary, uint256 value, uint256 timeToExpiration)` and call if from `hold` and `holdFrom`. So no code is duplicated. We can do this together, I think it's easier if I show you how.